SamplesDashboard/SamplesDashboard/ClientApp/src/components/Repositories/Repositories.tsx

```diff
@@ -0,0 +1,370 @@
+
+          // Sort the results by name by default
+          allRepositories.current = copyAndSort(
+            fetchedRepositories,
+            'name',
+            false
+          );
+
+          // Set the repos in the view
+          setRepositories(allRepositories.current);
+          // Calculate and set the stats
+          setRepositoryStats(getRepositoryStats(allRepositories.current));
+        } else {
+          app.displayError!(
+            'An error occurred while loading repositories',
+            `API returned ${response.status}`
+          );
+        }
+      } catch (error: any) {
+        app.displayError!(
+          'An error occurred while loading repositories',
+          error.toString()
+        );
+      }
+
+      setIsLoading(false);
+    };
+
+    loadRepositories();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [msal.instance]);
+
+  // Called as user types in the name text field
+  const onFilterByName = (
+    ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+    text?: string
+  ): void => {
+    setRepositories(
+      text
+        ? allRepositories.current.filter(
+            (r) => r.name.toLowerCase().indexOf(text.toLowerCase()) > -1
+          )
+        : allRepositories.current
+    );
+  };
+
+  // Called as user types in the owner text field
+  const onFilterByOwner = (
+    ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+    text?: string
+  ): void => {
+    setRepositories(
+      text
+        ? allRepositories.current.filter((r) => {
+            // Check each owner for the value in the text field
+            for (const owner of r.owners) {
+              if (
+                owner.displayName.toLowerCase().includes(text.toLowerCase())
+              ) {
+                return true;
+              }
+            }
+            return false;
+          })
+        : allRepositories.current
+    );
+  };
+
+  // Using useCallback here because this event handler
+  // needs to be updated every time the repositories state value changes
+  // otherwise, it always uses a stale copy
+  const onColumnClick = useCallback(
+    (ev: React.MouseEvent<HTMLElement>, column: IColumn): void => {
+      const newColumns = columns.current.slice();
+      const currentColumn = newColumns.filter(
+        (currCol) => column.key === currCol.key
+      )[0];
+      newColumns.forEach((newColumn) => {
+        if (newColumn === currentColumn) {
+          newColumn.isSortedDescending = !newColumn.isSortedDescending;
+          newColumn.isSorted = true;
+        } else {
+          newColumn.isSorted = false;
+          newColumn.isSortedDescending = true;
+        }
+      });
+
+      const newRepositories = copyAndSort(
+        repositories,
+        currentColumn.fieldName!,
+        currentColumn.isSortedDescending
+      );
+      columns.current = newColumns;
+      setRepositories(newRepositories);
+    },
+    [repositories]
```

Comment:
**suggestion (non-blocking)**: The concepts here look similar to the previous section `RepoDetail.tsx`. Consider creating a reusable function

Reply:
Well a lot of it is dealing with internal state of the component. However, I did move the processing of the column properties `isSorted` and `isSortedDescending` out into a separate reusable function. See the changes in https://github.com/microsoftgraph/msgraph-samples-dashboard/pull/417/commits/ce132dbce2c9b8accceae0143b657450fa6127f1.